---
title: "Report"
author: "Shane Ridoux"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    css: styles.css
    js: event.js
    toc: true
    pandoc_args: ["--lua-filter", "tabset.lua"]
    toc_float: 
      collapse: false
      smoothscroll: false
    df_print: kable
    code_folding: hide
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding) }) #, output_dir = "../reports") })
---
<!-- Markdown actually starts on line 48 -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
## Sets the maximum height of output boxes and makes them scrollable if necessary
options(width = 60)
local({
  hook_output <- knitr::knit_hooks$get('output')
  knitr::knit_hooks$set(output = function(x, options) {
    if (!is.null(options$max.height)) options$attr.output <- c(
      options$attr.output,
      sprintf('style="max-height: %s;"', options$max.height)
    )
    hook_output(x, options)
  })
})
```

```{cat, engine.opts=list(file = "tabset.lua")}
## calls tabset.lua file to make tabs
```

```{r clear_console, include = FALSE}
rm(list = ls())
cat("\014")
library(data.table)
library(table1)
library(infotheo)
library(igraph)
library(stringi)
library(dplyr)
library(ggplot2)
library(ggraph)
library(minet)
source("/Users/shane/School/CU-Denver/Masters-Project/masters-project/entropy.R")
source("/Users/shane/School/CU-Denver/Masters-Project/masters-project/information-gain.R")
set.seed(12)
```

<!-- Use this after header if you want tabs -->
<!-- {.tabset .tabset-fade .tabset-pills} -->
<!---------------- Start of Markdown ------------------>

# **OVERVIEW**
## Background
Type 1 Diabetes (T1D) is an autoimmune disorder where the immune system erroneously targets and destroys insulin-producing pancreatic islet-$\beta$ cells, leading to a lack of insulin and elevated blood glucose levels. Often diagnosed in early childhood, T1D left untreated can result in serious complications such as diabetic ketoacidosis, a life-threatening condition. With the disease incidence rising, efforts to understand the pathogenesis and etiology of T1D continue with the hopes of phase-specific therapeutic intervention to mitigate life-threatening complications, eliminate the burden of insulin pumps, and improve quality of life.

Type 1 Diabetes is strongly influenced by genetic factors, with nearly 40\% of the risk attributed to the highly polymorphic Major Histocompatibility Complex (MHC), which is associated with various diseases. This region includes genes encoding components of the complement system which acts as a rapid and targeted innate immune defense mechanism against pathogens, primarily through promoting inflammation and modulating the adaptive immune response. However, the efficiency and regulation of complement activity varies among individuals largely due to inherited genetic differences.

## Motivation
While genetic polymorphisms within the complement system significantly affect its activity and regulation, their role in immune activation and T1D progression remains underexplored due to the oversight of epistatic interactions. Genetic epistasis is most commonly understood as gene interaction where the contribution of one gene on the phenotypic outcome is dependent on genetic background (citation). The complotype is the total inherited set of genetic variants in complement genes (citation) and can be thought of as the genetic background for the complement system. 

The complotype impacts the activation potential of the complement and immune system where an overactive complement system influences susceptibility to inflammation and autoimmune disease while an underactive complement system results in increased risk for infection. While individual polymorphisms within the complotype have a minor impact on activation potential, their aggregate effect is believed to be greatly amplified either synergistically or antagonistically (citation?). Polymorphisms in complement genes have been shown to be associated with type 1 diabetes however, epistasis in complement genes has yet to be studied in type 1 diabetes.

A Genome-Wide Association Study (GWAS) tests millions of representative single nucleotide polymorphisms (SNPs) from a population in a linear framework where each SNP is tested for association with a phenotype individually (citation). Testing interactions among these SNPs in pairs or tuples of length $n$ drastically increase the already large number of hypotheses. A typical approach adjusting for multiple hypotheses in a GWAS would be to use Bonferoni correction for the determined number of independent SNPs in a genome with the standard significance threshold being set at $5\times 10^{-8}$. 

There are issues with this standard of practice. The standard genome-wide significance threshold faces challenges in evolving GWAS practices. It was developed for common variants, potentially lacking power for rare variants. It does not take into account the prior probability of a variant being associated with a phenotype, nor does it account for the statistical power of the test which both influence the interpretation of the p-value. Additionally, its foundation on the number of independent SNPs in a European reference population neglects the genetic diversity across populations and the influence of genetic context. These oversights can distort results, particularly when considering epistasis, where interactions between variants may not align with assumptions of independence. Applying the threshold universally risks missing meaningful associations while inflating false positives, especially in diverse populations or complex phenotypes. Bayesian methods have been applied to address false positives in GWAS especially in the context of epistasis.

# **PREPROCESSING**{.tabset .tabset-fade .tabset-pills}
## Quality Control
```{r, qc}
qc <- fread("/Users/shane/School/CU-Denver/Masters-Project/plink2/250120-GWAS_QC_notes.txt")
qc
```

The resulting Table 1 looks like:
```{r, table1}
gt <- as.data.frame(fread("/Users/shane/School/CU-Denver/Masters-Project/plink2/genotype_matrix.raw"))
pheno <- gt[,1:6]

pheno$SEX <- factor(pheno$SEX, labels = c("Male","Female"))
pheno$PHENOTYPE <- factor(pheno$PHENOTYPE, labels = c("Control","IA"))
label(pheno$SEX) <- "Sex"
label(pheno$PHENOTYPE) <- "IA Status"
table1(~ SEX | PHENOTYPE, pheno)
```

## Annotation

# **WITHIN-GENE SYNERGY**
The concept of bivariate synergism quantifies the combined effect of two SNPs ($A$ and $B$) on a disease ($D$) beyond their individual contributions using information theory. It is calculated as:  

$Syn(A; B; D) = I(A, B; D) -[I(A; D) + I(B; D)]$  

where $Syn(A; B; D)$ compares the joint contribution of SNPs $A$ and $B$ to the disease $D$ with the additive contributions of the individual SNPs. The information gain I(A; D) about the disease $D$ due to knowledge about SNP $A$ and is defined as:  

$I(A; D) = H(D) - H(D|A)$  
$I(A, B; D) = H(D) - H(D|A, B)$    where $H(\cdot)$ is the entropy.   
$H(D) = \sum_d p(d)log(\frac{1}{p(d)})$  
$H(D|A) = \sum_{a,d} p(a,d)log(\frac{1}{p(d|a)})$   where $p(d)$ is the probability of having disease $D=d$ and $p(d|a)$ is the probability of having disease $D=d$ given SNP $A$ has genotype $a$.  
A Network results from the bivariate synergy calculations of each SNP where the edge of nodes $A$ and $B$ has weight $Syn(A; B; D)$.

```{r, within-gene-synergy}
data <- fread("/Users/shane/School/CU-Denver/Masters-Project/corrected-pheno.txt")
load("/Users/shane/Library/CloudStorage/OneDrive-TheUniversityofColoradoDenver/Immunogenetics_T1D/data/complement/daisy/input/analysis_file_3levels.RData")

data <- merge(data, analysis_file, by.x = "IID", by.y = "ID")
data <- as.data.frame(data) # gets rid of data.table class



snps <- colnames(data)[21:27] # get snps
combos <- combn(snps, 2) # make combos of snps to test
self <- matrix(sort(rep(snps,2)),2) # make snpA - snpA pair to test
pairs <- matrix(c(combos,self),2) # combine combos with self rep snp pairs 
syn=c()
SNP1=c()
SNP2=c()
# compute bivariate synergy between snps given pheno
for(i in 1:dim(pairs)[2]){
  snps <- pairs[,i]
  bi <- synergy(data = data, pheno = "Phenotype_discretized", snps = snps)
  SNP1 = c(SNP1, snps[1])
  SNP2 = c(SNP2, snps[2])
  syn = c(syn, bi)
}
bisyn=data.frame(SNP1,SNP2,syn)

## Make a square matrix from the dataframe of bivariate synergy
# get names for row and columns
nameVals <- sort(unique(unlist(bisyn[1:2])))
# construct 0 matrix of correct dimensions with row and column names
myMat <- matrix(0, length(nameVals), length(nameVals), dimnames = list(nameVals, nameVals))
# fill in the matrix with matrix indexing on row and column names
myMat[as.matrix(bisyn[c("SNP1", "SNP2")])] <- bisyn[["syn"]]
myMat[as.matrix(bisyn[c("SNP2", "SNP1")])] <- bisyn[["syn"]]
LD<-myMat
diag(LD) = 0 #make diagonal zero i.e no info between the same snp
# print(LD)

## Make the Diffusion Laplacian matrix
#LD<-myMat 
D<-diag(rowSums(LD))
# print("D")
# print(D)
Laplacian<-as.matrix(D-LD)
print("Laplacian ===========================================")
print(Laplacian)   

gene<-density<-mean_dist<-transitivity<-edge_dens<-vertex_con<-edge_con<-NULL
  
############################################################
## select meaningfull edges using minet pkg and summarise graph
###########################################################

#--------------------------------------------------------------------------
# mrnet: Maximum Relevance Minimum Redundancy
graph_mrnet = mrnet(LD)
  
datgraph_mrnet = graph_from_adjacency_matrix(graph_mrnet, 
                                             mode = "undirected", 
                                             weighted = TRUE,
                                             diag = FALSE)
#remove loops
datgraph_mrnet = simplify(datgraph_mrnet,
                          remove.multiple=TRUE,
                          remove.loops=TRUE)
  
# gene[[file]] = file
# -----------------------------   summaries of interest
# density = edge_density(datgraph_mrnet,loop=FALSE)  #Density
  
# mean_dist = mean_distance(datgraph_mrnet)  #Average Path Length
  
# transitivity = transitivity(datgraph_mrnet)    #Clustering Coefficeint
  
# edge_dens = edge_density(datgraph_mrnet, loops=F) #number of edges/no.of posible edges
  
# vertex_con = vertex_connectivity(datgraph_mrnet) #number of edges/no.of posible edges
# edge_con = edge_connectivity(datgraph_mrnet) #number of edges/no.of posible edges

# snpbetw_centr = betweenness(datgraph_mrnet, directed=F, weights=NA)
# snpbetw_centr = data.frame(snpbetw_centr)
##dim(snpbetw_centr)
# snpsbetw_centrDF <- tibble::rownames_to_column(snpbetw_centr, "SNP")


# graphxx = data.frame(gene,density,mean_dist,transitivity,edge_dens,vertex_con,edge_con)
# graphxx = data.frame(density,mean_dist,transitivity,edge_dens,vertex_con,edge_con)
  


# graphxx = setNames(graphxx, c("gene", "density", "mean_dist", "transitivity", "edge_dens", "vertex_con", "edge_con"))
# graphxx = setNames(graphxx, c("density", "mean_dist", "transitivity", "edge_dens", "vertex_con", "edge_con"))
  
  
# Set node size by degree centrality
node_size <- degree(datgraph_mrnet,
                    mode = "all")
  
# Create a layout
graph_layout <- layout_with_fr(datgraph_mrnet)  # Force-directed layout
  
# Plot the network
ggraph(datgraph_mrnet, layout = graph_layout) +
  geom_edge_link(aes(edge_alpha = weight), show.legend = FALSE) +
  geom_node_point(aes(size = node_size), color = "steelblue", show.legend = FALSE) +
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +
  theme_void() +
  ggtitle("Within-Gene Synergy Network")
  

```

# **CONCLUSION**
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla facilisi. Fusce ac tellus et massa efficitur posuere. Sed vel convallis arcu. Nulla facilisi. Mauris ultricies est eu turpis molestie, nec dictum metus efficitur. Donec nec lacus sed ipsum posuere hendrerit. Duis sit amet metus vitae metus faucibus tempor nec a enim. Donec lobortis convallis nunc, id lacinia ante tempor in. Proin in libero at ipsum facilisis vehicula.

# **SESSION INFO**
```{r}
sessionInfo()
```

<!-- Footer -->

![](images/CampusMark_Dual_standard_c_cmyk-1-removebg-preview.png)
